packages/table/components/CheckboxTable.tsx

```diff
@@ -7,43 +8,74 @@ export interface CheckboxTableProps extends TableProps {
   /**
    * Callback for when a user makes a selection. Passes selected row data as a parameter
    */
-  onChange?: (selectedRows: object[]) => void;
+  onChange?: (selectedRows: {}) => void;
+  /**
+   * Selected row data
+   */
+  selectedRows?: {};
+  /**
+   * The unique key in each row's data object.
+   *
+   * Ideally, the value of this key in your table's data
+   * should be a number or string
+   */
+  uniqueKey: string;
 }
 
-class CheckboxTable extends React.PureComponent<CheckboxTableProps> {
+export interface CheckboxTableState {
+  headerChecked: boolean;
+}
+
+class CheckboxTable extends React.PureComponent<
+  CheckboxTableProps,
+  CheckboxTableState
+> {
   constructor(props) {
     super(props);
 
+    this.state = {
+      headerChecked: false
+    };
+
     this.getSelectedRows = this.getSelectedRows.bind(this);
     this.checkboxCellRenderer = this.checkboxCellRenderer.bind(this);
     this.getHeaderCheckbox = this.getHeaderCheckbox.bind(this);
   }
 
+  public componentWillReceiveProps(nextProps) {
```

Comment:
This method is deprecating and will disappear in React 17. It looks like you can replace it with getDerivedStateFromProps fairly easily

Reply:
Great point, I'll swap it out.